### PR TITLE
Tyle8552/packages

### PR DESCRIPTION
**Pull request by ghost**
This PR has a number of major updates:
1. Structured with packages instead of nested projects.
2. Removed large resources for offline data. These are downloaded from AGOL to a folder called `samples-data` into the project root with the gradle task `downloadData` the first time the samples are refreshed/built.
3. Using the new version of the gradle plugin to get the api dependency. It also downloads the jniLibs and resources into the project root directory. You don't have to configure an environment variable and no more `.arcgis` folder.

Without the nested projects, samples are run differently. You can right click the class file and run in an IDE, or you can specify the main class in the command line with

```
./gradlew run -PmainClass="com.esri.samples.scene.display_scene.DisplaySceneSample"
```

~~Changes to the gradle plugin must be merged before this can be merged.~~

### CODE REVIEW

**Comment by ghost** [comment]
Hey @john0005, lets chat about this when you get a chance.

**Comment by john0005** [comment]
Sounds good just let me know when you get in and we can do it before scrum or after.

**Comment by ghost** [comment]
@mbcoder This PR changes the samples from a multi project structure to a package structure.

File: build.gradle [hunk 1]
```diff
@@ -1,15 +1,46 @@
...
-        }
+        /*maven {
+            url 'https://esri.bintray.com/arcgis'
+        }*/
+        mavenLocal()
     }
-
     dependencies {
-        compile 'com.esri.arcgisruntime:arcgis-java:100.0.0-1450'
+        classpath 'com.esri.arcgisruntime:gradle-arcgis-java-plugin:1.0.0-SNAPSHOT'
```
Comment (doneill) [inline build.gradle:12]: Where are you pulling this from?  Dependency cannot be retrieved so import to IntelliJ fails.

**Comment by doneill** [comment]
@tyle8552 Although not explicitly against official naming conventions, I would reconsider the use of underscore in the package name, e.g. prefer  `com.esri.samples.scene.displayscene.DisplaySceneSample` to the naming you provided in your opening comment.

**Comment by ghost** [comment]
@doneill What about

```
com.esri.samples.featurelayers.servicefeaturetablemanualcache.ServiceFeatureTableManualCacheSample
```

vs.

```
com.esri.samples.featurelayers.service_feature_table_manual_cache.ServiceFeatureTableManualCacheSample
```

File: build.gradle (hunk 1 above)
Comment (ghost) [reply build.gradle:12]: @doneill It's coming from `mavenLocal()`. Waiting on that plugin pull request to be accepted.

**Comment by ghost** [comment]
@john0005 Everything should work now but only for build 1468. We can ask Dan to put a more recent build on artifactory (until this process is automated). I've also updated all of the sample READMEs to html.

**Comment by john0005** [comment]
Got it to work and the samples are running.
What structure looks like in eclipse:
newsamples

If you can update the doc on how to add a sample I will start switching over my samples to work with the new setup.
https://devtopia.esri.com/runtime/java-se-sdk/wiki/Adding-Samples

**Comment by ghost** [comment]
@john0005 I'd definitely recommend using the Project Explorer View for a hierarchical view. I'll update the doc.

**Comment by ghost** [comment]
@vGandhi Can you do a quick look over please?

**Comment by vGandhi** [comment]
@tyle8552 Looks good.
I skimmed through the commit history - in build.gradle, does the maven repo need to be reverted to bintray?

**Comment by ghost** [comment]
No, we'll keep it on artifactory to test with daily builds until it gets merged into master.

**Pull request merged**